### PR TITLE
added flanneld cluster component

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1559.

### CODE REVIEW

File: tls_assets.go [hunk 1]
```diff
@@ -33,18 +33,20 @@ const (
...
-	// WorkerComponent is a worker component.
-	WorkerComponent ClusterComponent = "worker"
-	// EtcdComponent is the etcd cluster component.
-	EtcdComponent ClusterComponent = "etcd"
 	// CalicoComponent is the calico component.
 	CalicoComponent ClusterComponent = "calico"
+	// EtcdComponent is the etcd cluster component.
+	EtcdComponent ClusterComponent = "etcd"
+	// FlanneldComponent is the flanneld cluster component.
+	FlanneldComponent ClusterComponent = "flanneld"
```
Comment (xh3b4sd) [inline tls_assets.go:41]: This got added.

File: tls_assets.go [hunk 2]
```diff
@@ -33,18 +33,20 @@ const (
 const (
 	// APIComponent is the API server component.
 	APIComponent ClusterComponent = "api"
-	// WorkerComponent is a worker component.
-	WorkerComponent ClusterComponent = "worker"
```
Comment (xh3b4sd) [inline tls_assets.go:37]: Boyscouting. Sorting alphabetically.

File: tls_assets.go [hunk 3]
```diff
@@ -70,12 +72,15 @@ type AssetsBundle map[AssetsBundleKey][]byte
 
 // ClusterComponents is a slice enumerating all the components that make up the
 // cluster.
+//
+// TODO add FlanneldComponent as soon as all guest clusters have Flanneld
+// certificates.
```
Comment (xh3b4sd) [inline tls_assets.go:77]: I want to do this later.

Comment (jgsqware) [reply tls_assets.go:77]: Why adding now the component?

Comment (xh3b4sd) [reply tls_assets.go:77]: The issue this work tries to solve in the end is to provide a separate key pair to the flannel daemons used in each guest cluster. This here is the very first step to get the reference for the component used to generate certificates. We can do it in a smooth way and just wait until all guest clusters have the key pair generated. Then we can actually use them and add the new component to the lookup when reconciling guest clusters.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**